Makefile

```diff
@@ -21,6 +21,7 @@ $(COMMON_CONSTRAINTS_TXT):
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: $(COMMON_CONSTRAINTS_TXT)
 	## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -qr requirements/pip-tools.txt
 	$(UPGRADE) --allow-unsafe --rebuild -o requirements/pip.txt requirements/pip.in
 	$(UPGRADE) -o requirements/pip-tools.txt requirements/pip-tools.in
 	pip install -qr requirements/pip-tools.txt
```

Comment:
maybe we could just move this line up?

Reply:
This actually (theoretically) needs to installed here, since what's happening is that we're using pip-tools to upgrade pip-tools, then want to use the upgraded version for the rest of the flow. There was an extra 3rd(!) install step below, though, that I've removed.